test/unit/util/sfdcUrlTest.ts

```diff
@@ -0,0 +1,218 @@
+      expect(protocol1).to.equal('http:');
+      expect(protocol2).to.equal('http:');
+      expect(emitWarningSpy.callCount).to.equal(1);
+    });
+  });
+
+  describe('getJwtAudienceUrl', () => {
+    const env = new Env();
+    before(() => {
+      $$.SANDBOX.stub(MyDomainResolver.prototype, 'getCnames').resolves(TEST_CNAMES);
+    });
+
+    it('return the jwt audicence url for sandbox domains', async () => {
+      const url = new SfdcUrl('https://organization.my.salesforce.com');
+      const response = await url.getJwtAudienceUrl();
+      expect(response).to.be.equal('https://test.salesforce.com');
+    });
+
+    it('return the jwt audicence url for internal domains (same)', async () => {
+      const url = new SfdcUrl('https://organization.stm.salesforce.com');
+      const response = await url.getJwtAudienceUrl();
+      expect(response).to.be.equal('https://organization.stm.salesforce.com');
+    });
+
+    it('return the jwt audicence url for sandbox domains', async () => {
+      const url = new SfdcUrl('https://organization.sandbox.my.salesforce.com');
+      const response = await url.getJwtAudienceUrl();
+      expect(response).to.be.equal('https://test.salesforce.com');
+    });
+
+    it('should use the correct audience URL for createdOrgInstance beginning with "gs1"', async () => {
+      $$.SANDBOX.stub(MyDomainResolver.prototype, 'getCnames').resolves([]);
+      const url = new SfdcUrl('https://foo.bar.baz');
+      const response = await url.getJwtAudienceUrl('gs1');
+      expect(response).to.be.equal('https://gs1.salesforce.com');
+    });
+
+    it('should return production URL if domain cannot be resolved', async () => {
+      $$.SANDBOX.stub(MyDomainResolver.prototype, 'getCnames').resolves([]);
+      const url = new SfdcUrl('https://foo.bar.baz');
+      const response = await url.getJwtAudienceUrl();
+      expect(response).to.be.equal('https://login.salesforce.com');
+    });
+
+    it('should use the correct audience URL for SFDX_AUDIENCE_URL env var', async () => {
+      env.setString('SFDX_AUDIENCE_URL', 'http://authInfoTest/audienceUrl/test');
+      const url = new SfdcUrl('https://login.salesforce.com');
+      const response = await url.getJwtAudienceUrl();
+      expect(response).to.be.equal(process.env.SFDX_AUDIENCE_URL);
+    });
+  });
+
+  describe('lookup', () => {
+    let resolveStub: SinonStub;
+    beforeEach(() => {
+      resolveStub = stubMethod($$.SANDBOX, MyDomainResolver.prototype, 'resolve').resolves(TEST_IP);
+    });
+
+    afterEach(() => {
+      resolveStub.restore();
+    });
+
+    it('should be able to do dns lookup', async () => {
+      const url = new SfdcUrl('https://foo.bar.baz');
+      const ip = await url.lookup();
+      expect(ip).to.be.equal(TEST_IP);
+    });
+
+    it('shoult throw on dns resulution failure', async () => {
+      resolveStub.restore();
+      resolveStub = stubMethod($$.SANDBOX, MyDomainResolver.prototype, 'resolve').rejects();
```

Comment:
Since there are just 2 tests for `lookup` and they each use a different stub I think we should remove the beforeEach and afterEach and just define the stubs in the tests.  That would simplify it a bit.